src/components/HairCheck/HairCheck.js

```diff
@@ -0,0 +1,128 @@
+import React, { useEffect, useRef, useCallback, useState } from 'react';
+import {
+  useLocalParticipant,
+  useVideoTrack,
+  useDevices,
+  useDaily,
+  useDailyEvent,
+} from '@daily-co/daily-react-hooks';
+import UserMediaError from '../UserMediaError/UserMediaError';
+
+import './HairCheck.css';
+
+export default function HairCheck({ joinCall, cancelCall }) {
+  const localParticipant = useLocalParticipant();
+  const videoTrack = useVideoTrack(localParticipant?.session_id);
+  const { microphones, speakers, cameras, setMicrophone, setCamera, setSpeaker } = useDevices();
+  const callObject = useDaily();
+  const videoElement = useRef();
+
+  const [getUserMediaError, setGetUserMediaError] = useState(false);
+
+  useDailyEvent(
+    'camera-error',
+    useCallback(() => {
+      setGetUserMediaError(true);
+    }, []),
+  );
+
+  const onChange = (e) => {
+    callObject.setUserName(e.target.value);
+  };
+
+  const join = (e) => {
+    e.preventDefault();
+    joinCall();
+  };
+
+  const cancel = () => {
+    cancelCall();
```

Comment:
does this need to be defined if it's just cancelCall? I might just pass cancelCall as the click handler instead.